File: data/java/AI/A3.java
import java.io.*;
import java.nio.file.*;
import java.util.*;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

/**
 * AI-generated log parser
 */
public class LogParserAI {

    private static final String LOG_DIR = "logs/";

    static class LogEntry {
        LocalDateTime timestamp;
        String level;
        String message;

        LogEntry(LocalDateTime ts, String lvl, String msg) {
            timestamp = ts;
            level = lvl;
            message = msg;
        }

        String toCSV() {
            return timestamp + "," + level + "," + message;
        }
    }

    // Parse single log line
    public static LogEntry parseLine(String line) {
        // Example format: [2025-08-31T12:34:56] INFO - Message text
        try {
            int first = line.indexOf("]");
            String tsStr = line.substring(1, first);
            LocalDateTime ts = LocalDateTime.parse(tsStr);
            String rest = line.substring(first + 2);
            String[] parts = rest.split(" - ", 2);
            return new LogEntry(ts, parts[0], parts[1]);
        } catch (Exception e) {
            return null;
        }
    }

    // Read all log files
    public static List<LogEntry> readLogs(String dir) throws IOException {
        List<LogEntry> entries = new ArrayList<>();
        DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(dir), "*.log");
        for (Path path : stream) {
            List<String> lines = Files.readAllLines(path);
            for (String line : lines) {
                LogEntry entry = parseLine(line);
                if (entry != null) entries.add(entry);
            }
        }
        return entries;
    }

    // Save logs to CSV
    public static void saveCSV(List<LogEntry> entries, String fileName) throws IOException {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
            writer.write("Timestamp,Level,Message\n");
            for (LogEntry entry : entries) {
                writer.write(entry.toCSV());
                writer.newLine();
            }
        }
    }

    public static void main(String[] args) {
        try {
            List<LogEntry> logs = readLogs(LOG_DIR);

            // Filter error logs
            List<LogEntry> errors = new ArrayList<>();
            for (LogEntry e : logs) {
                if (e.level.equals("ERROR")) errors.add(e);
            }

            saveCSV(logs, "all_logs.csv");
            saveCSV(errors, "error_logs.csv");

            System.out.println("Parsed " + logs.size() + " entries, errors: " + errors.size());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: data/js/AI/A3.js
// AI-generated logging utility
// Reads logs, filters errors, and saves to JSON

const fs = require('fs');
const path = require('path');

// Directory containing log files
const LOG_DIR = './logs';

// Read all log files from directory
function readLogs(dir) {
    const files = fs.readdirSync(dir);
    const logs = [];
    for (const file of files) {
        if (file.endsWith('.log')) {
            const content = fs.readFileSync(path.join(dir, file), 'utf-8');
            const lines = content.split(/\r?\n/);
            for (const line of lines) {
                const log = parseLine(line);
                if (log) logs.push(log);
            }
        }
    }
    return logs;
}

// Parse a log line into an object
function parseLine(line) {
    // Example: [2025-08-31T12:34:56] INFO - Message
    const match = line.match(/\[(.*?)\] (\w+) - (.*)/);
    if (!match) return null;
    return { timestamp: match[1], level: match[2], message: match[3] };
}

// Save logs to JSON file
function saveJSON(data, file) {
    fs.writeFileSync(file, JSON.stringify(data, null, 2));
    console.log(`Saved ${data.length} entries to ${file}`);
}

// Main processing
function main() {
    const logs = readLogs(LOG_DIR);
    const errors = logs.filter(l => l.level === 'ERROR');

    saveJSON(logs, 'all_logs.json');
    saveJSON(errors, 'error_logs.json');

    console.log(`Processed ${logs.length} logs, ${errors.length} errors`);
}

main();
